### ISSUE TITLE
New mod testing guide creation

### ISSUE THREAD

**Issue by scooby78** [scooby78/160thSOR#54]

**Comment by yettie79** [comment]
WIP with Diznutz doing questions

**Comment by yettie79** [comment]
Members are expressing frustration with the limitations of mods, they feel its to hard to get anything added. 
this had been expressed to me by a number of members and with enough frustration that they are threatening to leave the community.

**Comment by scooby78** [comment]
Yeah I know it must be frustrating..

As discussed recently, with the stabilization of ARMA and surrounding mods plus the removal of unused mods in our pack, steps moving forward with this have already started.

I think the fundamental idea we must make sure is known is more mods means more problems after ARMA updates / possible mod conflicts / performance degradation and the more important problem of our core foundation may begin to change too much (training, gameplay / mission balancing & inter battalion cooperation) as fun as they may be, some mods are just not practical to add to the official mod pack. 

Are Friday night missions being made too regimented? 
They are the ‘free for all’ and ‘any thing goes’ missions, are they being still used for that?

These mods can still be (and I’m cautious to add a size limit here) added to the server and run on Fridays or whenever to allow members to still use a mod with others and go nuts.  This can actually promote incidental rigorous testing at the same time as being fun..

**Comment by Nels0** [comment]
The only regulation of Friday nights is that you have to reserve them on the calendar, and make an announcement post. Free-for-all is definitely encouraged, but I'm unsure how well known it is that mods can be added

**Comment by Nels0** [comment]
I suggest we loosen up on mods and mod testing slightly. I volunteer to be a point of contact for people, upload/manage mods, keep track of tests etc. This may slightly overlap with @Deek so let me know if I'm stepping on toes here.

**Comment by yettie79** [comment]
Just remember any mod that is being tested for the pack must be tested with only 160th mods.

Nelso, thanks for the offer however i feel you have a shit load to do as head of missions

**Comment by scooby78** [comment]
@Nels0  as much as I understand where you are coming from..  What is your reasoning for suggesting to loosen up on mods and mod testing?

**Comment by Nels0** [comment]
The issue of adding mods is probably better discussed in person, but in a nutshell I am fairly convinced that we need to add more variety to operations. More mods is not the only way to do this but will help greatly, and also inspire mission makers to do more exciting things.
We don't necessarily need to lower our standards, more ensure that due process is followed _quickly._

**Comment by scooby78** [comment]
Your reasoning is sound and quite frankly I don’t disagree..  In fact we are developing the opfor pack to do just that.  

The issue lays with the cost of adding in such mods on stability / performance and additional training (if needed).

Regardless of how fun / cool or practical the mod may be, many have previously not passed scrutiny or (and more often the case) the hype or interest has only lasted a few missions.  An example of two mods that HAVE passed this scrutiny would be Advanced movement and sling loading.  They are used countless times AND needed extra training.  

I pose you these historical events and questions:

FFIS AI mod was implemented with many tests performed with small groups of players.  It was fantastic but when inserted, caused a conflict with ACE creating a insta-death on first bullet hit to players at random.  This damaged our player base as it took time and considerable effort to figure out that is was FFIS.

This is a lesson learned over and over spanning more than 4 years.

> If we relax the addition of any mod and lower the scrutiny threshold, what safeguard do you suggest to mitigate conflicts?

With the addition of mods this is increasing our mod pack size.  When I started, the pack was 16Gb it steadily grew to almost 30Gb there at one stage but currently sits around 22Gb.  For new members to be expected to download such a large amount.

> If we add additional mods that are requested, when do you suggest we cap the size of the pack?

> Currently, we are not using all of the assets we currently have in our pack, what percentage of usage do you suggest we accept for wasted space in our pack?

With the fact that ARMA is evolving with DLC and updates do bring optimization’s and new assets.  When APEX came out there where many assets that you needed the DLC to play the mission.    Missions became unplayable to members that didn’t have the DLC (even for one little camo net that they where never going to see).  APEX also optimized the way Guns worked, it was great, but it broke every gun mod out there.  Some could argue the fact that you COULD just click past the error, they would be right.  Dedicated servers don’t follow that logic unfortunately.  Massive efforts by 5 people yielded a result that allowed the missions to continue as planned when I know of other communities that came to a grinding halt.  If the knowledge and passion from all involved was  not available we would have stopped for 2 weeks also.

> Assuming we add all these mods, what safeguard do you suggest for protecting the community from the unexpected conflicts of future ARMA updates?

Since we have members that aren’t rich enough to afford rental time of a quantum computer to play ARMA, some mods are going to effect the frames on their computer.  “So what! Just tell them to lower their view distance” is something I’ve heard a lot.  In some cases this may be the solution.  I know of players that are already down to minimum and still being effected.  “But that’s their problem, not mine!” Is  usually the blunt comment in retort, and to a degree they probably make a good logical argument...  But they are wrong... the dedicated server works hard to sync everyone up to the same sequence of events so what you see is what you get yeah?  Nope! What you see is what YOU see and what you GET is a negotiation with your computer, the server, the other players inside draw distance, Zeus’s computer and the outcome from the server and Zeus’s computer.  This is a lot of talking that happens A LOT. Pick just 1 link in this negotiation and make it ‘late to the party’, sure the information gets there but like most negotiations, new information can restart a debate! Around we go again!! Thankfully this is happening in the millisecond realm.  BUT those renegotiations stop everything else from progressing so you get (if everyone is agreeing often) good frames, (if some are a little late) slow frames and (if some are really late) rubber banding and really slow frames.  But all of this can be caused by about 10-20 different elements that don’t even include the mod in question!!  Which one is causing the problem? Or worse which ONES are?

Adding a new mod risks this very unstable balance, which CAN be disrupted by one slow computer struggling with a mod.

> If we add any mod that is cool, how do you suggest we validate the poor performance in a mission when we don’t know who is struggling or what actually caused the issue for weeks after the mission failed to perform?

> How do you suggest we get new players to return when this is their first experience with the community?

> How do you suggest  we stop them talking to their friends and telling them that frames are poor in missions?

In summary, the above is really only the tip of the iceberg for me.  I really want certain mods myself, and they are really cool!  But they just aren’t good enough to insert or only serve a small portion of the community..  sometimes..  I use my own words against me, and I can’t honestly validate the cost to the rest of the community.

The addition of mods has a far reaching and a continuing burden to the community, the question is and always will be “Is the cost worth it ..... Really?”. 

And that is what the new testing guide is being developed to help with.

**Comment by Nels0** [comment]
I understand the basics of Arma networking, and I know that mods can break things. Again, I am saying we should not lower our standards. 
Given that we have been told Sunday missions are for testing, I would like to see them used for that more often, and so would the other mission makers. 
I understand the risk of scrapped missions and large downloads for players. 
I firmly believe that the community is in need of some fresh content, and not just on Friday and Saturday nights.
This mod testing guide is the right way to go, and we need to ensure the execution is followed all the way through, with support and constructive criticism from all members involved.

**Comment by Deek356** [comment]
Mod Testing Document as at 19/11/17
https://docs.google.com/document/d/120-sCgdea8r7c0zZiwXR2eNDcaJAWiAEDj9WRViQHGY/edit

**Comment by Deek356** [comment]
Zeus Mod & Map test report
https://docs.google.com/forms/d/1Bs09eqJiNSM9x3l00xTqdGonLh_q6z0aPRG-KtOnQJk/edit

160th Mod & Map Testing Form
https://docs.google.com/forms/d/1JnHbnq5gcjrydTMFNq6Fg6iXP6hN8VZKEYYtaVAZgWY/edit
